src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -0,0 +1,912 @@
+            }
+
+            if (frameworkType == typeof(BinaryData))
+            {
+                return format is SerializationFormat.Bytes_Base64 or SerializationFormat.Bytes_Base64Url
+                    ? BinaryDataExpression.FromBytes(element.GetBytesFromBase64(format.ToFormatSpecifier()))
+                    : BinaryDataExpression.FromString(element.GetRawText());
+            }
+
+            if (IsCustomJsonConverterAdded(frameworkType) && serializationType is not null)
+            {
+                return InvokeJsonSerializerDeserializeMethod(element, serializationType);
+            }
+
+            if (frameworkType == typeof(JsonElement))
+                return element.InvokeClone();
+            if (frameworkType == typeof(object))
+                return element.GetObject();
+            if (frameworkType == typeof(bool))
+                return element.GetBoolean();
+            if (frameworkType == typeof(char))
+                return element.GetChar();
+
+            if (frameworkType == typeof(short))
+                return element.GetInt16();
+            if (frameworkType == typeof(int))
+                return element.GetInt32();
+            if (frameworkType == typeof(long))
+                return element.GetInt64();
+            if (frameworkType == typeof(float))
+                return element.GetSingle();
+            if (frameworkType == typeof(double))
+                return element.GetDouble();
+            if (frameworkType == typeof(decimal))
+                return element.GetDecimal();
+            if (frameworkType == typeof(string))
+                return element.GetString();
+            if (frameworkType == typeof(Guid))
+                return element.GetGuid();
+            if (frameworkType == typeof(byte[]))
+                return element.GetBytesFromBase64(format.ToFormatSpecifier());
+
+            if (frameworkType == typeof(DateTimeOffset))
+            {
+                return format == SerializationFormat.DateTime_Unix
+                    ? DateTimeOffsetExpression.FromUnixTimeSeconds(element.GetInt64())
+                    : element.GetDateTimeOffset(format.ToFormatSpecifier());
+            }
+
+            if (frameworkType == typeof(DateTime))
+                return element.GetDateTime();
+            if (frameworkType == typeof(TimeSpan))
+            {
+                if (format == SerializationFormat.Duration_Seconds)
+                {
+                    return TimeSpanExpression.FromSeconds(element.GetInt32());
+                }
+
+                if (format == SerializationFormat.Duration_Seconds_Float)
+                {
+                    return TimeSpanExpression.FromSeconds(element.GetDouble());
+                }
+
+                return element.GetTimeSpan(format.ToFormatSpecifier());
+            }
+
+            throw new NotSupportedException($"Framework type {frameworkType} is not supported");
```

Comment:
There are a few RPs failing in the regen. I did not go through all of them, but one of them ([log](https://dev.azure.com/azure-sdk/590cfd2a-581c-4dcb-a12e-6568ce786175/_apis/build/builds/3101210/logs/266)) is complaining this:
Exception: Framework type Azure.ResourceManager.Resources.Models.ExtendedLocationType is not supported
Considering the existence of our customization code, there are possibilities that we could put any framework type in the model, and we should not throw here.
Here is the model: https://github.com/Azure/azure-sdk-for-net/blob/5d5acaabce95ab28faee3c1eb6d164bf7b7d9442/sdk/resourcemanager/Azure.ResourceManager/src/Resources/Generated/Models/ExtendedLocationType.cs#L14
it is an extensible enum.
How do we previously handle this type?

Reply:
Well there are 4 errors in total.
2 of them are reporting `ExtendedLocationType`
the other 2 are reporting `RegionType` which is also an extensible enum from `Azure.ResourceManager` assembly.
They should share the same root cause.

Reply:
Yes. These types have custom deserialization methods. I need to merge the pre-build check from consolidation (or maybe relax the condition for type representation in expression tree).

Reply:
Fixed